ngraph/test/type_prop/acos.cpp

```diff
@@ -0,0 +1,61 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, acos_basic_shape_inference)
```

Comment:
Can we create a base test class for all unary arithmetic functions and use it everywhere (for asin, acos, asinh, acosh, atan, atanh etc)? This will allow us to avoid a lot of code duplication. This class should be parametrized with the type of operation, input tensor type and input shape.

Reply:
Sounds like a great idea! Will discuss it with a team